packages/webapp/src/components/Finances/Transaction/ExpandedContent/index.jsx

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { BsChevronRight } from 'react-icons/bs';
+import TextButton from '../../../Form/Button/TextButton';
+import history from '../../../../history';
+import { transactionTypeEnum } from '../../../../containers/Finances/useTransactions';
+import styles from './styles.module.scss';
+
+// TODO LF-3748, 3749, 3761
+const components = {
+  EXPENSE: (props) => <div>expense placeholder</div>,
+  REVENUE: (props) => <div>revenue placeholder</div>,
+  LABOUR_EXPENSE: (props) => <div>labour placeholder</div>,
+  CROP_SALE: (props) => <div>crop sale placeholder</div>,
+};
+
+const getDetailPageLink = ({ transactionType, relatedId }) => {
+  return {
+    LABOUR_EXPENSE: '/labour',
+    EXPENSE: `/expense/${relatedId}`,
+    REVENUE: `/revenue/${relatedId}`,
+  }[transactionType];
+};
+
+export default function ExpandedContent({ data }) {
+  const { typeLabel, transactionType } = data;
+
+  const { t } = useTranslation();
+
+  const toDetailText =
+    transactionType === transactionTypeEnum.labourExpense
+      ? t('FINANCES.TRANSACTION.VIEW_DETAILS')
+      : t('FINANCES.TRANSACTION.VIEW_AND_EDIT');
+
+  const componentKey = typeLabel === 'Crop Sale' ? 'CROP_SALE' : transactionType;
+
+  return (
+    <div className={styles.expandedContent}>
+      <TextButton className={styles.toDetail} onClick={() => history.push(getDetailPageLink(data))}>
+        {toDetailText}
+        <BsChevronRight />
+      </TextButton>
+      {components[componentKey]({ data })}
```

Comment:
Since the components object stores actual React functional components, it's always recommendable to render them as opposed to calling them like regular functions. Calling them as functions can cause problems if there are any hooks in the components (there's more detailed info/examples in this article https://kentcdodds.com/blog/dont-call-a-react-function-component).

So here I'd recommend doing this instead:

```js
const Component = components[componentKey];

return (
    <div>
        ...
        <Component {...data} />
    </div>
);
```